src/extension.ts

```diff
@@ -255,6 +257,12 @@ export async function activate(context: vscode.ExtensionContext) {
 
         recordToolkitInitialization(activationStartedOn, getLogger())
 
+        fs.watch(join(SystemUtilities.getHomeDirectory(), '.aws'), async (eventType, filename) => {
+            if ((filename === 'credentials' || filename === 'config') && eventType === 'change') {
```

Comment:
Should be calling `getCredentialsFilename` and `getConfigFilename` for the file paths as to respect the user's environment variables. Note that these functions return the full file paths. So either use two file watchers, or check if they share the same directory (which is the most likely case) and use a single watcher.